build.rs

```diff
@@ -173,52 +241,131 @@ fn create_ss58_registry(json: &str) -> Result<proc_macro2::TokenStream, String>
 		}
 
 		/// All non-custom address formats (Sorted by network)
-		static ALL_SS58_ADDRESS_FORMATS: [Ss58AddressFormatRegistry; #count] = [
+		pub(crate) static ALL_SS58_ADDRESS_FORMATS: [Ss58AddressFormatRegistry; #count] = [
 			 #(Ss58AddressFormatRegistry::#identifier),*,
 		];
 
 		/// Names of all address formats (Sorted by network)
-		static ALL_SS58_ADDRESS_FORMAT_NAMES: [&str; #count] = [
+		pub(crate) static ALL_SS58_ADDRESS_FORMAT_NAMES: [&str; #count] = [
 			#(#name),*,
 		];
 
 		/// (Sorted) prefixes to index of ALL_SS58_ADDRESS_FORMATS
-		static PREFIX_TO_INDEX: [(u16, usize); #count] = [
+		pub(crate) static PREFIX_TO_INDEX: [(u16, usize); #count] = [
 			#(#prefix_to_idx),*,
 		];
 
+		impl Ss58AddressFormatRegistry {
+			/// Tokens used on the given network.
+			pub fn tokens(&self) -> &'static[TokenRegistry] {
+				match self {
+					#(Ss58AddressFormatRegistry::#identifier => &[#ident_to_tokens],)*
+
+				}
+			}
+
+		}
+
 		impl Ss58AddressFormat {
 			/// Network/AddressType is reserved for future use.
 			pub fn is_reserved(&self) -> bool {
-				self.prefix > 16384 || matches!(self.prefix, #(#reserved_prefixes)|*)
+				self.prefix() > 16384 || matches!(self.prefix(), #(#reserved_prefixes)|*)
 			}
 
 			/// A custom format is one that is not already known.
 			pub fn is_custom(&self) -> bool {
 				// A match is faster than bin search
 				// as most hits will be in the first group.
-				!matches!(self.prefix, #(#prefix_starts..=#prefix_ends)|*)
+				!matches!(self.prefix(), #(#prefix_starts..=#prefix_ends)|*)
+			}
+		}
+
+
+		#[non_exhaustive]
+		#[doc = "List of all tokens used on some network in the ecosystem."]
```

Comment:
```suggestion
		/// List of all tokens used on some network in the ecosystem.
```

Not sure why you want to use the doc attribute here?